src/query/expression/src/evaluator.rs

```diff
@@ -103,6 +118,18 @@ impl<'a> Evaluator<'a> {
         #[cfg(debug_assertions)]
         self.check_expr(expr);
 
+        // try get result from cache
+        if let Some(cached_values) = &self.cached_values {
+            let found = cached_values.read().contains_key(expr);
```

Comment:
Why not use `cached_values.read().get` directly?

Reply:
I also found this case, in the previous logical if check false then perform a write to the cache. Fixed.